Narrower data type and sign conversion warnings in qt_sinks.h

```c++
include/spdlog/sinks/qt_sinks.h: In instantiation of ‘void spdlog::sinks::qt_color_sink<Mutex>::sink_it_(const spdlog::details::log_msg&) [with Mutex = spdlog::details::null_mutex]’:
include/spdlog/sinks/qt_sinks.h:150:10:   required from here
  150 |     void sink_it_(const details::log_msg &msg) override {
      |          ^~~~~~~~
include/spdlog/sinks/qt_sinks.h:163:71: error: conversion to ‘qsizetype’ {aka ‘long long int’} from ‘size_t’ {aka ‘long unsigned int’} may change the sign of the result [-Werror=sign-conversion]
  163 |                 color_range_start = QString::fromUtf8(str.data(), msg.color_range_start).size();
      |                                                                   ~~~~^~~~~~~~~~~~~~~~~
include/spdlog/sinks/qt_sinks.h:163:94: error: conversion from ‘qsizetype’ {aka ‘long long int’} to ‘int’ may change value [-Werror=conversion]
  163 |                 color_range_start = QString::fromUtf8(str.data(), msg.color_range_start).size();
      |                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
include/spdlog/sinks/qt_sinks.h:164:69: error: conversion to ‘qsizetype’ {aka ‘long long int’} from ‘size_t’ {aka ‘long unsigned int’} may change the sign of the result [-Werror=sign-conversion]
  164 |                 color_range_end = QString::fromUtf8(str.data(), msg.color_range_end).size();
      |                                                                 ~~~~^~~~~~~~~~~~~~~
include/spdlog/sinks/qt_sinks.h:164:90: error: conversion from ‘qsizetype’ {aka ‘long long int’} to ‘int’ may change value [-Werror=conversion]
  164 |                 color_range_end = QString::fromUtf8(str.data(), msg.color_range_end).size();
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~

```

Sorry for the delay.

```
gcc (GCC) 14.2.0 20240801 (OpenMandriva)
```

With Clang:

```
OpenMandriva 19.1.6-1 clang version 19.1.6 (/builddir/build/BUILD/llvm-19.1.6-build/llvm-project-llvmorg-19.1.6/clang 1cc1c8b2641ccba94c7b90b5e4c4e500729f7556)
```

```c++
include/spdlog/sinks/qt_sinks.h:163:71: error: implicit conversion changes signedness: 'size_t' (aka 'unsigned long') to 'qsizetype' (aka 'long long') [-Werror,-Wsign-conversion]
  163 |                 color_range_start = QString::fromUtf8(str.data(), msg.color_range_start).size();
      |                                     ~~~~~~~                       ~~~~^~~~~~~~~~~~~~~~~
include/spdlog/sinks/qt_sinks.h:164:69: error: implicit conversion changes signedness: 'size_t' (aka 'unsigned long') to 'qsizetype' (aka 'long long') [-Werror,-Wsign-conversion]
  164 |                 color_range_end = QString::fromUtf8(str.data(), msg.color_range_end).size();
      |                                   ~~~~~~~                       ~~~~^~~~~~~~~~~~~~~
include/spdlog/sinks/qt_sinks.h:163:71: error: implicit conversion changes signedness: 'size_t' (aka 'unsigned long') to 'qsizetype' (aka 'long long') [-Werror,-Wsign-conversion]
  163 |                 color_range_start = QString::fromUtf8(str.data(), msg.color_range_start).size();
      |                                     ~~~~~~~                       ~~~~^~~~~~~~~~~~~~~~~
include/spdlog/sinks/qt_sinks.h:68:5: note: in instantiation of member function 'spdlog::sinks::qt_color_sink<spdlog::details::null_mutex>::sink_it_' requested here
   68 |     qt_color_sink(QTextEdit *qt_text_edit,
      |     ^
include/spdlog/sinks/qt_sinks.h:164:69: error: implicit conversion changes signedness: 'size_t' (aka 'unsigned long') to 'qsizetype' (aka 'long long') [-Werror,-Wsign-conversion]
  164 |                 color_range_end = QString::fromUtf8(str.data(), msg.color_range_end).size();
      |                                   ~~~~~~~                       ~~~~^~~~~~~~~~~~~~~
include/spdlog/sinks/qt_sinks.h:174:45: error: implicit conversion changes signedness: 'const level::level_enum' to 'size_type' (aka 'unsigned long') [-Werror,-Wsign-conversion]
  174 |                              colors_.at(msg.level),  // color to apply
      |                                      ~~ ~~~~^~~~~
include/spdlog/sinks/qt_sinks.h:163:90: error: implicit conversion loses integer precision: 'qsizetype' (aka 'long long') to 'int' [-Werror,-Wshorten-64-to-32]
  163 |                 color_range_start = QString::fromUtf8(str.data(), msg.color_range_start).size();
      |                                   ~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
include/spdlog/sinks/qt_sinks.h:164:86: error: implicit conversion loses integer precision: 'qsizetype' (aka 'long long') to 'int' [-Werror,-Wshorten-64-to-32]
  164 |                 color_range_end = QString::fromUtf8(str.data(), msg.color_range_end).size();
      |                                 ~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
```

--- a/include/spdlog/sinks/qt_sinks.h
+++ b/include/spdlog/sinks/qt_sinks.h
@@ -160,8 +160,8 @@ protected:
             payload = QString::fromUtf8(str.data(), static_cast<int>(str.size()));
             // convert color ranges from byte index to character index.
             if (msg.color_range_start < msg.color_range_end) {
-                color_range_start = QString::fromUtf8(str.data(), msg.color_range_start).size();
-                color_range_end = QString::fromUtf8(str.data(), msg.color_range_end).size();
+                color_range_start = QString::fromUtf8(str.data(), static_cast<qsizetype>(msg.color_range_start)).size();
+                color_range_end = QString::fromUtf8(str.data(), static_cast<qsizetype>(msg.color_range_end)).size();
             }
         } else {
             payload = QString::fromLatin1(str.data(), static_cast<int>(str.size()));
@@ -171,7 +171,7 @@ protected:
                              qt_text_edit_,          // text edit to append to
                              std::move(payload),     // text to append
                              default_color_,         // default color
-                             colors_.at(msg.level),  // color to apply
+                             colors_.at(static_cast<size_t>(msg.level)),  // color to apply
                              color_range_start,      // color range start
                              color_range_end};       // color range end
 

